localstack/services/apigateway/provider.py

```diff
@@ -436,7 +444,7 @@ def put_method(
         # TODO: add missing validation? check order of validation as well
         moto_backend = apigw_models.apigateway_backends[context.account_id][context.region]
         moto_rest_api: MotoRestAPI = moto_backend.apis.get(rest_api_id)
-        if not moto_rest_api or not moto_rest_api.resources.get(resource_id):
+        if not moto_rest_api or not (moto_resource := moto_rest_api.resources.get(resource_id)):
```

Comment:
probably not needed.

Reply:
I'm using the resource under by getting the `moto_http_method` (line 489)